internal/os_updater/downloader.go

```diff
@@ -0,0 +1,19 @@
+/*
+ * SPDX-FileCopyrightText: (C) 2025 Intel Corporation
+ * SPDX-License-Identifier: LicenseRef-Intel
+ */
+
+// Package osupdater updates the OS.
+package osupdater
+
+// IDownloader is an interface that contains the method to download the update.
+type IDownloader interface {
```

Comment:
go convention is to omit the I in front of the interface name

Reply:
ok, I saw it on a Go example, so I included it.  I know we usually don't do that.